cms/static/js/views/course_video_settings.js

```diff
@@ -0,0 +1,537 @@
+
+        getFidelityPlan: function() {
+            var fidelityPlan = null;
+            if (this.selectedProvider === 'Cielo24') {
+                fidelityPlan = this.availableTranscriptionPlans[this.selectedProvider].fidelity;
+            }
+            return fidelityPlan;
+        },
+
+        getPlanLanguages: function() {
+            var selectedPlan = this.availableTranscriptionPlans[this.selectedProvider];
+            if (this.selectedProvider === 'Cielo24') {
+                return selectedPlan.fidelity[this.selectedFidelityPlan].languages;
+            }
+            return selectedPlan.languages;
+        },
+
+        fidelitySelected: function(event) {
+            var $fidelityContainer = this.$el.find('.transcript-fidelity-wrapper');
+            this.selectedFidelityPlan = event.target.value;
+            // Remove any error if present already.
+            this.clearPreferanceErrorState($fidelityContainer);
+
+            // Clear active and selected languages.
+            this.selectedLanguages = this.activeLanguages = [];
+            this.populateLanguages();
+        },
+
+        turnaroundSelected: function(event) {
+            var $turnaroundContainer = this.$el.find('.transcript-turnaround-wrapper');
+
+            this.selectedTurnaroundPlan = event.target.value;
+            // Remove any error if present already.
+            this.clearPreferanceErrorState($turnaroundContainer);
+        },
+
+        providerSelected: function(event) {
+            this.resetPlanData();
+            this.selectedProvider = event.target.value;
+            this.populatePreferences();
+        },
+
+        languageSelected: function(event) {
+            var $parentEl = $(event.target.parentElement).parent(),
+                $languagesEl = this.$el.find('.transcript-languages-wrapper'),
+                selectedLanguage = $parentEl.find('select').val();
+
+            // Remove any error if present already.
+            this.clearPreferanceErrorState($languagesEl);
+
+            // Only add if not in the list already.
+            if (selectedLanguage && _.indexOf(this.selectedLanguages, selectedLanguage) === -1) {
+                this.selectedLanguages.push(selectedLanguage);
+                this.addLanguage(selectedLanguage);
+                // Populate language menu with latest data.
+                this.populateLanguageMenu();
+            } else {
+                this.addErrorState($languagesEl);
+            }
+        },
+
+        languageRemoved: function(event) {
+            var selectedLanguage = $(event.target).data('language-code');
+            $(event.target.parentElement).parent().remove();
+
+            // Remove language from selected languages.
+            this.selectedLanguages = _.without(this.selectedLanguages, selectedLanguage);
+
+            // Populate menu again to reflect latest changes.
+            this.populateLanguageMenu();
+        },
+
+        populateProvider: function() {
```

Comment:
populateProvider  --> `renderProviders`. Looks like the purpose of this method is to render the radio buttons with values not populating the values in some existing elements.